bson/src/test/unit/org/bson/codecs/VectorCodecTest.java

```diff
@@ -0,0 +1,162 @@
+        assertNotEquals(actualDecodedDocument, expectedDocument);
+        Vector actualVector = binaryVector.asVector();
+        assertEquals(actualVector, vectorToEncode);
+    }
+
+    @ParameterizedTest
+    @MethodSource("provideVectorsAndCodecsForRoundTrip")
+    void shouldEncodeVector(final Vector vectorToEncode, final Codec<Vector> vectorCodec) {
+        // given
+        BsonWriter mockWriter = Mockito.mock(BsonWriter.class);
+
+        // when
+        vectorCodec.encode(mockWriter, vectorToEncode, EncoderContext.builder().build());
+
+        // then
+        verify(mockWriter, times(1)).writeBinaryData(new BsonBinary(vectorToEncode));
+        verifyNoMoreInteractions(mockWriter);
+    }
+
+    @ParameterizedTest
+    @MethodSource("provideVectorsAndCodecsForRoundTrip")
+    void shouldDecodeVector(final Vector vectorToDecode, final Codec<Vector> vectorCodec) {
+        // given
+        BsonReader mockReader = Mockito.mock(BsonReader.class);
+        BsonBinary bsonBinary = new BsonBinary(vectorToDecode);
+        when(mockReader.peekBinarySubType()).thenReturn(BsonBinarySubType.VECTOR.getValue());
+        when(mockReader.readBinaryData()).thenReturn(bsonBinary);
+
+        // when
+        Vector decodedVector = vectorCodec.decode(mockReader, DecoderContext.builder().build());
+
+        // then
+        assertNotNull(decodedVector);
+        assertEquals(vectorToDecode, decodedVector);
+    }
+
+
+    @ParameterizedTest
+    @EnumSource(value = BsonBinarySubType.class, mode = EnumSource.Mode.EXCLUDE, names = {"VECTOR"})
+    void shouldThrowExceptionForInvalidSubType(final BsonBinarySubType subType) {
+        // given
+        BsonReader mockReader = Mockito.mock(BsonReader.class);
+        when(mockReader.peekBinarySubType()).thenReturn(subType.getValue());
+
+        Stream.of(new Float32VectorCodec(), new Int8VectorCodec(), new PackedBitVectorCodec())
+                .forEach(codec -> {
+                    // when & then
+                    BSONException exception = assertThrows(BSONException.class, () ->
+                            codec.decode(mockReader, DecoderContext.builder().build()));
+                    assertEquals("Unexpected BsonBinarySubType", exception.getMessage());
+                });
+    }
+
+
+    @ParameterizedTest
+    @MethodSource("provideVectorsAndCodecsForRoundTrip")
+    void shouldReturnCorrectEncoderClass(final Vector vector, final Codec<? extends Vector> codec, final Class<? extends Vector> expectedEncoderClass) {
+        // when
+        Class<? extends Vector> encoderClass = codec.getEncoderClass();
+
+        // then
+        assertEquals(expectedEncoderClass, encoderClass);
+    }
+
+    @ParameterizedTest
+    @MethodSource("provideVectorsCodec")
+    void shouldConvertToString(final Codec<Vector> codec) {
+        // when
+        String result = codec.toString();
+
+        // then
+        assertEquals(codec.getClass().getSimpleName() + "{}", result);
+    }
```

Comment:
It's probably the first `toString` test I've seen. I don't think we need it even if the new vector codecs need to override `toString`, but on top of that, I don't think they should do that, as other codecs usually don't.